### DISCUSSION TITLE
Issue with selecting SOM clusters/metaclusters in the pixel clustering notebook

### DISCUSSION THREAD

**Discussion by honzee** [angelolab/ark-analysis#1032] [General]
Hi,

firstly, thank you for publishing such an interesting package. We are working on analysis of multiplex immunofluorescence images from COMET and we would be excited to try out your approach for the analysis.

I have segmented an ROI of our image using mesmer and used it as input for the pixel clustering notebook. The jupyter notebook ran well until the GUI part, where I got: Error displaying widget: model not found. 
The versions of ipywidgets was: 8.0.7 and jupyterlab_widgets was: 3.0.8

I downgraded to 7.7.0 and 1.1.0 respectively, which I knew would work. Unfortunately, selecting the clusters now does not work. When I click the specified regions, nothing happens. I can still rename the metaclusters though.

I would like to ask, whether you have some minimum version of these packages/ or environment requirements for the selection to work.

Best regards
Jan

**Comment by ngreenwald** [comment]
Hi Jan, 
If you're having dependency issues you can use the [docker container](https://github.com/angelolab/ark-analysis#setting-up-docker) to make sure you're using the correct versions of everything. 

I don't think those are pinned in the repo since they're installed in the docker. @srivarra should be able to comment on the best way to get that running within a conda environment.

**Comment by srivarra** [comment]
@honzee

Hey Jan,

We have a Conda `environment.yml` file which sets the user dependencies for the pipeline.

In your terminal run the following commands to clone the repo and create the environment.

```shell
git clone -b v0.6.4 https://github.com/angelolab/ark-analysis.git
cd ark-analysis
conda env create -f environment.yml
```

With the `environment.yml` the version of `ipywidgets` is 8.1.0, and`jupyterlab-widgets` is 3.0.8.

In addition the conda environment installs extra notebook dependencies grouped under `lab_ext` in the [`pyproject.toml`](https://github.com/angelolab/ark-analysis/blob/b09107e3de49a3acf3177e418a5dde0da011f76f/pyproject.toml#L75) file

Let us know if this solves the interactivity issue with the metaclustering GUI.

**Comment by honzee** [comment]
Thank you both for the help. I actually installed ark-analysis with conda as you suggested but got ipywidgets 8.0.7, which did not work and produced error: Error displaying widget: model not found
That is why I reverted to the older version I knew that works.
I will try to reinstall make sure the ipywidgets are 8.1.0 and let you know if that solves the problem
Best,
Jan

**Comment by honzee** [comment]
So I reinstalled the conda environment, with ipywidgets version 8.1.0, but still getting: Error displaying widget: model not found. Probably it is an environment issue. Currently I cannot run the jupyter in docker, otherwise that would be the best option of course.

**Comment by srivarra** [comment]
@honzee 

What operating system are you using? For example I am able to use the GUI on an Apple Silicon Mac.
I'm wondering if it could be an OS issue.

**Reply by honzee** [reply]
The jupyterlab is running on CentOS Linux 7 on a cluster, since we need the computational resources for analyzing the big images. However, I can try running it locally.
